tests/std/tests/P0896R4_ranges_alg_generate_n/test.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+struct instantiator {
+    template <ranges::output_range<const int&> Out>
+    static constexpr void call() {
+        using ranges::generate_n, ranges::equal, ranges::iterator_t;
+
+        const auto iota_gen = [count = 0]() mutable { return count++; };
+
+        {
+            int output[] = {13, 42, 1367};
+            Out out_wrapper{output};
+            auto result = generate_n(out_wrapper.begin(), ranges::distance(output), iota_gen);
+            STATIC_ASSERT(same_as<decltype(result), iterator_t<Out>>);

```

Comment:
I observe that the `generate` test isn't inspecting the iterator type returned by the range overload, like this `generate_n` test is doing. Not sure if they should be consistent.

Reply:
> Not sure if they should be consistent.

It certainly can't hurt.